src/main/java/dev/waterdog/waterdogpe/player/ProxiedPlayer.java

```diff
@@ -795,24 +796,48 @@ public Long2LongMap getEntityLinks() {
         return this.entityLinks;
     }
 
-    public PacketHandler getPluginUpstreamHandler() {
-        return this.pluginUpstreamHandler;
+    /**
+     * This method is deprecated. Please use {@link #getPluginUpstreamHandlers()} instead.
+     */
+    @Deprecated()
+    public List<PacketHandler> getPluginUpstreamHandler() {
```

Comment:
For backward compatibility, this should return first `PacketHandler`, and not the whole list. In the next version, this can be removed then.